elastic/src/main/java/org/openmbee/mms/elastic/CommitElasticDAOImpl.java

```diff
@@ -18,29 +18,74 @@
     }
 
     public void indexAll(Collection<? extends BaseJson> jsons) {
-        this.indexAll(getIndex(), associateCommits(jsons));
+        this.indexAll(getIndex(), jsons);
     }
 
     public void index(BaseJson json) {
-        if (json.get(CommitJson.COMMITID).toString().isEmpty()) {
-            CommitJson commitJson = (CommitJson) json;
-            commitJson.put(CommitJson.COMMITID, UUID.randomUUID().toString());
-        }
-        this.index(getIndex(), json);
+        index((CommitJson) json);
     }
 
-    public void index(BaseJson json, String commitId) {
-        this.index(getIndex(), addCommitId(json, commitId));
+    public void index(CommitJson json) {
+        if (json.getCommitId().isEmpty()) {
+            json.setCommitId(UUID.randomUUID().toString());
+        }
+
+        int commitCount = getCommitSize(json);
+        List<CommitJson> broken = new ArrayList<>();
+        if (commitCount > indexLimit) {
+            List<Map<String, Object>> allActions = new ArrayList<>();
+            allActions.addAll(json.getAdded().stream().peek(toAdd -> toAdd.put("action", "added")).collect(Collectors.toList()));
+            allActions.addAll(json.getUpdated().stream().peek(toUpdate -> toUpdate.put("action", "updated")).collect(Collectors.toList()));
+            allActions.addAll(json.getDeleted().stream().peek(toDelete -> toDelete.put("action", "deleted")).collect(Collectors.toList()));
+
+            while (!allActions.isEmpty()) {
+                CommitJson currentCommitCopy = new CommitJson();
+                currentCommitCopy.setCommitId(json.getCommitId());
+                currentCommitCopy.setSource(json.getSource());
+                currentCommitCopy.setCreated(json.getCreated());
+                currentCommitCopy.setCreator(json.getCreator());
+                currentCommitCopy.setModified(json.getModified());
+                currentCommitCopy.setModifier(json.getModifier());
+                currentCommitCopy.setAdded(new ArrayList<>());
+                currentCommitCopy.setUpdated(new ArrayList<>());
+                currentCommitCopy.setDeleted(new ArrayList<>());
+
+                while (getCommitSize(currentCommitCopy) < indexLimit && !allActions.isEmpty()) {
+                    Map<String, Object> action = allActions.remove(0);
+                    switch (action.getOrDefault("action", "none").toString()) {
+                        case "added":
+                            currentCommitCopy.getAdded().add(action);
+                            break;
+                        case "updated":
+                            currentCommitCopy.getUpdated().add(action);
+                            break;
+                        case "deleted":
+                            currentCommitCopy.getDeleted().add(action);
+                            break;
+                    }
+                }
+                broken.add(currentCommitCopy);
+
+            }
+            this.indexAll(broken);
```

Comment:
Just fyi, this adds a property called "action" into the commit objects. We could remove it if it's a problem.